script.js

```diff
@@ -135,11 +135,27 @@ function initListeners() {
     if (target.classList.contains("options__trash")) {
       let str = target.closest(".notes__item");
       str = str.querySelector(".notes__descr").textContent;
-      console.log(str);
+      // console.log(str);
       let updatedObj = notesObj.filter((note) => note.note !== str);
       notesObj = updatedObj;
     }
 
+    // check note
+    if (
+      target.classList.contains("notes__item-check") ||
+      target.tagName === "IMG"
+    ) {
+      let str = target.closest(".notes__item");
+      str = str.querySelector(".notes__descr").textContent;
+      // console.log(str);
+
+      notesObj.forEach((note) => {
+        if (note.note === str) {
+          note.isChecked = !note.isChecked; // toggle checked state
```

Comment:
Problem 3: Changing note data in place
You wrote:

```js
note.isChecked = !note.isChecked;
```

Why it's bad:
It changes the object directly, which can cause bugs in larger apps.
Not good if you're using frameworks like React later.

Better way:
Return a new object when updating:

```js
notesObj = notesObj.map(note =>
  note.note === noteText ? { ...note, isChecked: !note.isChecked } : note
);
```